plugin.idea/src/com/microsoft/alm/plugin/idea/common/setup/LinuxStartup.java

```diff
@@ -49,31 +46,65 @@ public static void startup() {
      * Create .desktop file to register the protocol handler
      *
      * @param scriptFile
-     * @return script file
      * @throws IOException
      */
-    protected static File createDesktopFile(final File scriptFile, final File desktopFile) throws IOException {
-        final FileWriter fileWriter = new FileWriter(desktopFile);
-        final BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+    protected static void createDesktopFile(final File scriptFile, final File desktopFile) throws IOException, InterruptedException {
```

Comment:
Since createDesktopFile does more work now (checks for changes and performs the local desktop database update), you may want to consider updating the function name.  Maybe updateDesktopFileAndDatabase?